File: Codeforces/Free_Problem_Solving/domino_laying/my_solution.cpp
// https://codeforces.com/problemset/problem/50/A

#include <iostream>

using namespace std;

int main(){

    int m,n;
    cin >> m >> n;

    if (m % 2 == 0 && n % 2 == 0){
        cout << (m / 2) * (n / 1) << endl;
    } else if (m % 2 == 0 && n % 2 != 0) {
        cout << (m / 2) * n << endl;
    } else if (m % 2 != 0 && n % 2 != 0) {
        cout << m * (n / 2) + (n % 2) * (m / 2) << endl;
    } else if (n % 2 == 0 && m % 2 != 0){
        cout << m * (n / 2) << endl;
    }

    return 0;
}

File: Codeforces/Free_Problem_Solving/beautiful_matrix/my_solution.py
# https://codeforces.com/problemset/problem/263/A

def make_beutiful_matrix(ugly_eye):
    beautiful_eye = (2,2)
    return abs(beautiful_eye[0] - ugly_eye[0]) + abs(beautiful_eye[1] - ugly_eye[1])


eye_position = (0,0)
for i in range(5):
    row_of_matrix = [int(i) for i in input().split()]

    if 1 in row_of_matrix:
        eye_position = (i, row_of_matrix.index(1))


print(make_beutiful_matrix(eye_position))


